plugins/genesys/fu-genesys-usbhub-device.c

```diff
@@ -748,27 +800,32 @@ fu_genesys_usbhub_device_setup(FuDevice *device, GError **error)
 				   (guint8 *)&self->static_ts,
 				   sizeof(FuGenesysStaticToolString));
 	}
-	if (self->static_ts.tool_string_version != 0xff) {
-		gchar rev[3] = {0};
+	if (memcmp(&self->static_ts.tool_string_version, "USB", 3) == 0) {
+		g_set_error(error,
+			    FWUPD_ERROR,
+			    FWUPD_ERROR_INTERNAL,
+			    "Unsupported IC: tool string unsupported");
```

Comment:
I would prefer `"Static Tool String not supported"` or `"Static Tool String %s not supported", (const gchar *)self->static_ts` to stay consistent with other error message.

Reply:
As the [comment](https://github.com/fwupd/fwupd/pull/4426/files#r835114733).
How about `String Descriptor of Tool not supported`?

Reply:
>  How about String Descriptor of Tool not supported?

I wanted to specify that it is the *Static Tool String* (i.e. not Dynamic, not Firmware, not Vendor...).

I guess, your suggestion makes sense if the check is done in the  function `fu_genesys_usbhub_device_get_descriptor_data()`.

Therefore, if it is correct to check for the "USB" prefix for Static, Dynamic, Firmware, Vendor TS... go ahead and move the `if (memcmp(...))` to `fu_genesys_usbhub_device_get_descriptor_data()`, that looks better to my eyes. Does it make sense?

Reply:
For me `"Static Tool String not supported"` makes more sense; because it checks for the static.

But if you prefer `"Tool String not supported"` or `"String Descriptor of Tool not supported"` go ahead.

I tend to stay consistent when I can; ~~Unsupported IC~~ and ~~unsupported~~ -> not supported;